+ klasy patient, patientController, patientRepository, patientService

Cymek12 opened the pull request.

Cymek12 committed:
+ klasy patient, patientController, patientRepository, patientService

konradgab commented on src/main/java/com/example/demo/controller/PatientController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @GetMapping("/{email}")
+    public Patient getPatientByEmail(@PathVariable("email") String email){
```
ctrl alt l

konradgab commented on src/main/java/com/example/demo/controller/PatientController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @PostMapping
+    public void addNewPatient(@RequestBody Patient patient){
```
ctrl alt l

konradgab commented on src/main/java/com/example/demo/controller/PatientController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @GetMapping
+    public List<Patient> getPatients(){
```
ctrl alt l

konradgab commented on src/main/java/com/example/demo/controller/PatientController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @DeleteMapping("/{email}")
+    public void deletePatientByEmail(@PathVariable("email") String email){
```
ctrl alt l

konradgab commented on src/main/java/com/example/demo/controller/PatientController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @PutMapping("/{email}")
+    public boolean editPatient(@PathVariable("email") String email, @RequestBody Patient newPatient) {
```
ctrl alt l

konradgab commented on src/main/java/com/example/demo/model/Patient.java:
```diff
@@ -0,0 +1,103 @@
+    private String firstName;
+    private String lastName;
+    private Long phoneNumber;
+    private String birthday;
```
LocalDate

konradgab commented on src/main/java/com/example/demo/model/Patient.java:
```diff
@@ -0,0 +1,103 @@
+    private Long idCardNo;
+    private String firstName;
+    private String lastName;
+    private Long phoneNumber;
```
String

konradgab commented on src/main/java/com/example/demo/model/Patient.java:
```diff
@@ -0,0 +1,103 @@
+public class Patient {
+    private String email;
+    private String password;
+    private Long idCardNo;
```
String

konradgab commented on src/main/java/com/example/demo/repository/PatientRepository.java:
```diff
@@ -0,0 +1,29 @@
+    private List<Patient> patients;
+
+    public PatientRepository(List<Patient> patients) {
+        this.patients = new ArrayList<>();
```
this.patients = patients;

konradgab commented on src/main/java/com/example/demo/repository/PatientRepository.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    public List<Patient> getPatients() {
+        return patients;
```
zwracamy kopie

konradgab commented on src/main/java/com/example/demo/repository/PatientRepository.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    public Optional<Patient> getPatientByEmail(String email){
+        return patients.stream().filter(patient -> patient.getEmail().equals(email)).findAny();
```
formatowanie streama

konradgab replied:
return patients.stream()
.filter(patient -> patient.getEmail().equals(email))
.findAny();

konradgab commented on src/main/java/com/example/demo/repository/PatientRepository.java:
```diff
@@ -0,0 +1,29 @@
+public class PatientRepository {
+    private List<Patient> patients;
+
+    public PatientRepository(List<Patient> patients) {
```
where usuniecie i edycja

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+
+@Service
+public class PatientService {
+    private PatientRepository patientRepository;
```
final

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+
+    public Patient getPatientByEmail(String email) {
+        Optional<Patient> optionalPatient = patientRepository.getPatientByEmail(email);
+        if (optionalPatient.isPresent()) {
```
to co napisales to jest dokladnie optionalPatient.get();

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public Patient getPatientByEmail(String email) {
+        Optional<Patient> optionalPatient = patientRepository.getPatientByEmail(email);
```
Patient patient = patientRepository.getPatientByEmail(email)
.orElseThrow(() -> new PatientNotFoundException ("nie ma pacjenta"))

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public void addNewPatient(Patient patient) {
+        patientRepository.addNewPatient(patient);
```
addPatient

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+        Optional<Patient> optionalPatient = patientRepository.getPatientByEmail(email);
+        if (optionalPatient.isPresent()) {
+            Patient patient = optionalPatient.get();
+            patientRepository.getPatients().remove(patient);
```
SKANDAL

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public boolean deletePatientByEmail(String email) {
+        Optional<Patient> optionalPatient = patientRepository.getPatientByEmail(email);
```
Patient patient = patientRepository.getPatientByEmail(email)
.orElseThrow(() -> new PatientNotFoundException());
patientRepository.delete(patient);

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public boolean editPatient(String email, Patient newPatient) {
+        Optional<Patient> optionalPatient = patientRepository.getPatientByEmail(email);
```
Patient patient = patientRepository.getPatientByEmail(email)
.orElseThrow(() -> new PatientNotFoundException);
patientRepository.update(patient, newData)

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -0,0 +1,59 @@
+
+    public boolean editPatient(String email, Patient newPatient) {
+        Optional<Patient> optionalPatient = patientRepository.getPatientByEmail(email);
+        if(optionalPatient.isPresent()){
```
ctrl alt l

Cymek12 committed:
zmiana typow pol w Patient,
zmiana nazewnictwa metod,
+ PatientNotFoundException
+ operacje CRUD do PatientRepository
wywalenie niepotrzebnych optionali z PatientService

Cymek12 committed:
edycja metody addPatient, walidacja przy dodawaniu pacjenta

Crasheroo commented on src/main/java/com/example/demo/repository/PatientRepository.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    public void editPatient(Patient patient, Patient newPatientData) {
+        patient.setEmail(newPatientData.getEmail());
```
A co jak pacjenta nie ma

Cymek12 committed:
+ GlobalExceptionHandler

Cymek12 committed:
+ dodanie adnotacji @ControllerAdvice

konradgab commented on src/main/java/com/example/demo/handler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,23 @@
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
```
MedicalClinicExceptionHandler

konradgab commented on src/main/java/com/example/demo/handler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,23 @@
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(value
+            = { PatientAlreadyExistException.class, PatientNotFoundException.class })
```
@ResponseStatus(NOT_FOUND)

Cymek12 committed:
+ ErrorMessage
zmiana klasy GlobalExceptionHandler na MedicalClinicExceptionHandler

Cymek12 committed:
Zastosowanie adnotacji z Lombok'a aby usunąć boiler code

konradgab commented on src/main/java/com/example/demo/model/ErrorMessage.java:
```diff
@@ -0,0 +1,12 @@
+import java.util.Date;
+
+@Builder
+
```
pusta linia

konradgab commented:
approved

Cymek12 committed:
usuniecie pustej lini
